src/gui/sharelinkwidget.cpp

```diff
@@ -154,23 +154,24 @@ void ShareLinkWidget::setupUiOptions()
     permissionsGroup->setExclusive(true);
 
     if (_isFile) {
-        checked = perm & (SharePermissionRead & SharePermissionUpdate);
+        checked = (perm & SharePermissionRead) && (perm & SharePermissionUpdate);
         _allowEditingLinkAction = _linkContextMenu->addAction(tr("Allow Editing"));
         _allowEditingLinkAction->setCheckable(true);
         _allowEditingLinkAction->setChecked(checked);
 
     } else {
-        checked = perm & SharePermissionRead;
+        checked = (perm == SharePermissionRead);
         _readOnlyLinkAction = permissionsGroup->addAction(tr("Read only"));
         _readOnlyLinkAction->setCheckable(true);
         _readOnlyLinkAction->setChecked(checked);
 
-        checked = perm & (SharePermissionRead & SharePermissionCreate & SharePermissionUpdate & SharePermissionDelete);
+        checked = (perm & SharePermissionRead) && (perm & SharePermissionCreate)
+            && (perm & SharePermissionUpdate) && (perm & SharePermissionDelete);
         _allowUploadEditingLinkAction = permissionsGroup->addAction(tr("Allow Upload && Editing"));
```

Comment:
Oh I guess someone earlier went crazy with all the bitwise stuff, to also use it in the action text `Allow Upload && Editing` 😂

This would be a great opportunity to correct that typo too* ;-)

(* the only thing I fear is, this would perhaps break translations as our release is imminent shorthand?)

Reply:
I seriously had not realized the && :dagger: 
Fixed it now. The wording matches the server.

Reply:
> Oh I guess someone earlier went crazy with all the bitwise stuff, to also use it in the action text `Allow Upload && Editing` joy

Unrelated I think, it's because labels use & for the mnemonics, so you need to repeat it twice if you want a & to appear: https://doc.qt.io/qt-5/qlabel.html#setBuddy